### ISSUE TITLE
highlight-regexp doesn't work in ledger-mode ?

### ISSUE THREAD

**Issue by simonmichael** [ledger/ledger-mode#53]
I just tried using highlight-regexp to highlight "TODO" in yellow, but it doesn't work unless I turn off font-lock-mode. That's a pity, and as far as I know unusual, that they won't coexist ?

**Comment by purcell** [comment]
The "TODO" was placed in comments, presumably?

**Comment by simonmichael** [comment]
It can also be in description or account names, eg.

**Comment by jstamant** [comment]
This behavior isn't happening for me (6 years later).

I came across this issue while submitting a pull-request for fixing the comment highlighting for postings.

When I use `highlight-regexp` (with `font-lock-mode` of course) to highlight all occurences of `TODO`, it successfully highlights them all, regardless of where they are. If `TODO` is found _anywhere_ in the file, it gets highlighted over the fontification specified by ledger-mode. I've tried it in tags, in directives, in transaction lines, in posting lines, globally, etc.

Running Emacs version 29.1, and the latest ledger-mode version from MELPA (version 4.0.0?)
